TicTacToe/TicTacToe.cs

```diff
@@ -1,87 +1,173 @@
+               Console.WriteLine("Your test is {0}", test);
+            } else {
+                Console.WriteLine ("Your test is {0}", test);
+            }
         }
-
-        public static void GetInput()
-        {
-            Console.WriteLine("Player " + playerTurn);
-            Console.WriteLine("Enter Row:");
-            int row = int.Parse(Console.ReadLine());
-            Console.WriteLine("Enter Column:");
-            int column = int.Parse(Console.ReadLine());
+        //getInput fx takes user input and uses it to fill board and call PlaceMark
+        public static void GetInput () {
+            Console.WriteLine ("Player " + playerTurn);
+            Console.WriteLine ("Enter Row:");
+            int row = int.Parse (Console.ReadLine ());
+            Console.WriteLine ("Enter Column:");
+            int column = int.Parse (Console.ReadLine ());
+            PlaceMark (row, column);
+            playerTurn = (playerTurn == "X") ? "O" : "X";
         }
-
-        public static void PlaceMark(int row, int column)
-        {
-        // your code goes here
+        //this will reference the board and check for input, if input found prompts to try again.
+        public static string PlaceMark (int row, int column) {
+            //X is default using ternary operator
+            if (board[row][column] == " ") {
+                board[row][column] = playerTurn;
+                return playerTurn;
+            } else if (board[row][column] == "X" || board[row][column] == "O") {
+                Console.WriteLine ("This space is taken...try again.");
+                GetInput ();
+                //ternary opporator changes the player.
+                playerTurn = (playerTurn == "X") ? "O" : "X";
+            }
+            return playerTurn;
         }
-
-        public static bool CheckForWin()
-        {
+        //checks all win conditions for a win.
+        public static bool CheckForWin () {
             // your code goes here
-
-            return false;
+            if (HorizontalWin () == true) {
+                return true;
+            } else if (VerticalWin () == true) {
+                return true;
+            } else if (DiagonalWin () == true) {
+                return true;
+            } else {
+                return false;
+            }
         }
-
-        public static bool CheckForTie()
-        {
+        //looks at the number of spaces - player keep playing until you win or there are no spaces left and you tie.
+        public static bool CheckForTie () {
             // your code goes here
-
-            return false;
+            int spaceAvail = 9;
+            for (int i = 0; i < board.Length; i++) {
+                for (int j = 0; j < board[i].Length; j++) {
+                    if (board[i][j] == "X" || board[i][j] == "O") {
+                        spaceAvail--;
+                    }
+                }
+            }
+            if (spaceAvail <= 0 && !CheckForWin () == true) {
```

Comment:
good catch. Checking for available space alone, is not enough to know if there is a tie!